### PR TITLE
Properly consider component parameter type

### LINKED ISSUE #1753: bash operator properties - boolean value is not handled properly
**Describe the issue**
Likely not specific to this operator and a general problem:

**To Reproduce**
1. Create AA RTC
2. Create AA pipeline
3. Add bash operator
4. Configure required properties without changing anything else
5. Note that xcom checkbox is selected by default
   image
6. Save the pipeline
7. Open the pipeline in a text editor and locate the `bashoperator_xcom_push` property. It is set to `false`

image

### PR DESCRIPTION
**Pull request by lresende**
Fixes #1753

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Review by kiersten-stokes — APPROVED** [review]
LGTM! Working as expected

File: elyra/pipeline/registry.py [hunk 1]
```diff
@@ -523,6 +523,17 @@ def parse_component_properties(self, component_body, component_path):
 
                     # Add to existing parameter list
                     component_parameters['parameters'].append({"id": new_parameter_info['parameter_ref']})
+                    if(default_value is not None and len(default_value) > 0):
```
Comment (kevin-bates) [inline elyra/pipeline/registry.py:526]: This could probably be simplified to: `if default_value:`.  If `False` or `0`, it should do the right thing, but that would be where the issue might lie.  But if its going to be a string regardless, then its save to simplify the conditional.

File: elyra/pipeline/registry.py [hunk 2]
```diff
@@ -523,6 +523,17 @@ def parse_component_properties(self, component_body, component_path):
...
+                        if 'data' in new_parameter_info:
```
Comment (kevin-bates) [inline elyra/pipeline/registry.py:527]: Should there be an immediate check against `default_value == 'None'` to set `default_value = None`, _then_ go through the data-type checks, etc.?

File: elyra/pipeline/registry.py [hunk 3]
```diff
@@ -523,6 +523,17 @@ def parse_component_properties(self, component_body, component_path):
...
+                            if 'format' in new_parameter_info['data']:
+                                component_parameter_format = new_parameter_info['data']['format']
+                                if (component_parameter_format):
```
Comment (kevin-bates) [inline elyra/pipeline/registry.py:530]: The parenthesis are redundant and some linters will flag as an error.

File: elyra/pipeline/registry.py [hunk 4]
```diff
@@ -523,6 +523,17 @@ def parse_component_properties(self, component_body, component_path):
...
+                                    if component_parameter_format == 'bool' or component_parameter_format == 'Boolean':
+                                        default_value = default_value in ('True', 'true')
```
Comment (kevin-bates) [inline elyra/pipeline/registry.py:532]: Should there be an else to set to `False` if not `None`?  (see previous None comment)

**Comment by lresende** [comment]
Thanks for the feedback @kevin-bates, I have updated the pr and also identified a few corner cases, and added those to the test cases.

**Review by kevin-bates — APPROVED** [review]
Looks good - great tests!
